---
title: "AMCSeqAnalysis"
author: "BYDavis"
date: "2024-02-29"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Markdown file to analyze AMC water eDNA samples, using Dr. Sue Ishaq's DNA Sequencing Analysis workflow (to start). 

Debugging statements are preceded with ###

# Install and load relevant packages
```{r packageinstall}
if(!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("dada2")
install.packages("ggplot2")
install.packages("beepr")

```

```{r loadpackages}
library(dada2)
library(ggplot2)
library(beepr)
```

Prior to this Rmd, the raw fastq files have been renamed in the repo UM_FSM_Cleaning

# Set folder paths and file types
```{r raw_paths, echo = FALSE}
# For the 12S samples:

# Set the file paths for the raw fastqs
path_raw12F <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/12S/Read1"
path_raw12R <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/12S/Read2"

# Specify file name formats
fns12F <- list.files(path_raw12F)
fastqs12F <- fns12F[grepl('.gz$', fns12F)]

fns12R <- list.files(path_raw12R)
fastqs12R <- fns12R[grepl('.gz$', fns12R)]

###  print(fastqs12F)

# For the BF2 samples:

# Set the file paths for the raw fastqs
path_rawBF <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/BF2BR2/Read1"
path_rawBR <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/BF2BR2/Read2"

# Specify file name formats
fnsBF <- list.files(path_rawBF)
fastqsBF <- fnsBF[grepl('.gz$', fnsBF)]

fnsBR <- list.files(path_rawBR)
fastqsBR <- fnsBR[grepl('.gz$', fnsBR)]

# The file paths still include the desktop.ini file, so let's specify a file path to lead future code only to the fastq files in the folder

fns12Fisolate <- file.path(path_raw12F, fastqs12F)
fns12Risolate <- file.path(path_raw12R, fastqs12R)

fnsBFisolate <- file.path(path_rawBF, fastqsBF)
fnsBRisolate <- file.path(path_rawBR, fastqsBR)

# Set sample names to a vector
# Remove path and the .gz extension
names12Ffast <- tools::file_path_sans_ext(basename(fastqs12F))
# Repeat file_path_sans_ext to also remove the .fastq
names12F <- tools::file_path_sans_ext(names12Ffast)

names12Rfast <- tools::file_path_sans_ext(basename(fastqs12R))
names12R <- tools::file_path_sans_ext(names12Rfast)

namesBFfast <- tools::file_path_sans_ext(basename(fastqsBF))
namesBF <- tools::file_path_sans_ext(namesBFfast)

namesBRfast <- tools::file_path_sans_ext(basename(fastqsBR))
# Repeat file_path_sans_ext to also remove the .fastq
namesBR <- tools::file_path_sans_ext(namesBRfast)

# Set location to put output images and graphs
path_outputs <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs"
```

The sample names are already set how I want them, so I won't use/adapt Sue's file name cleaning chunk (Clean File Names for Initial Load)

# Check Sequence Quality

Run 4 randomly selected quality profile plots per primer per read type (forward or reverse). Number will correspond to the file order. Min is set to 2 to avoid the desktop.ini file that GoogleDrive desktop creates from being chosen

```{r random_selection}

# Numbers for 12S Forward reads:
sample(1:78, 4, replace = FALSE)
  # Results on 3/5/2023: 41, 49, 18, 60

# Numbers for 12S Reverse reads:
sample(1:78, 4, replace = FALSE)
  # Results on 3/5/2023: 31, 3, 22, 55

# Numbers for BF2 Forward reads:
sample(1:78, 4, replace = FALSE)
  # Results on 3/5/2023: 70, 28, 62, 3

# Numbers for BF2 Reverse reads:
sample(1:78, 4, replace = FALSE)
  # Results on 3/5/2023: 10, 63, 49, 48

```


```{r qualityplotgeneration}
# 12S Forward:
plotQualityProfile(fns12Fisolate[41])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ForwardQuality_RNG41.png", plot = plotQualityProfile(fns12Fisolate[41]))

plotQualityProfile(fns12Fisolate[49])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ForwardQuality_RNG49.png", plot = plotQualityProfile(fns12Fisolate[49]))

plotQualityProfile(fns12Fisolate[18])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ForwardQuality_RNG18.png", plot = plotQualityProfile(fns12Fisolate[18]))

plotQualityProfile(fns12Fisolate[60])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ForwardQuality_RNG60.png", plot = plotQualityProfile(fns12Fisolate[60]))


# 12S Reverse:
plotQualityProfile(fns12Risolate[31])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ReverseQuality_RNG31.png", plot = plotQualityProfile(fns12Risolate[31]))

plotQualityProfile(fns12Risolate[3])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ReverseQuality_RNG3.png", plot = plotQualityProfile(fns12Risolate[3]))

plotQualityProfile(fns12Risolate[22])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ReverseQuality_RNG22.png", plot = plotQualityProfile(fns12Risolate[22]))

plotQualityProfile(fns12Risolate[55])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12ReverseQuality_RNG55.png", plot = plotQualityProfile(fns12Risolate[55]))


# B Forward:
plotQualityProfile(fnsBFisolate[70])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ForwardQuality_RNG70.png", plot = plotQualityProfile(fnsBFisolate[70]))

plotQualityProfile(fnsBFisolate[28]) 
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ForwardQuality_RNG28.png", plot = plotQualityProfile(fnsBFisolate[28]))

plotQualityProfile(fnsBFisolate[62])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ForwardQuality_RNG62.png", plot = plotQualityProfile(fnsBFisolate[62]))

plotQualityProfile(fnsBFisolate[5])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ForwardQuality_RNG5.png", plot = plotQualityProfile(fnsBFisolate[5]))


# B Reverse:
plotQualityProfile(fnsBRisolate[10])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ReverseQuality_RNG10.png", plot = plotQualityProfile(fnsBRisolate[10]))

plotQualityProfile(fnsBRisolate[63]) 
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ReverseQuality_RNG63.png", plot = plotQualityProfile(fnsBRisolate[63]))

plotQualityProfile(fnsBRisolate[49])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ReverseQuality_RNG49.png", plot = plotQualityProfile(fnsBRisolate[49]))

plotQualityProfile(fnsBRisolate[48])
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF2ReverseQuality_RNG48.png", plot = plotQualityProfile(fnsBRisolate[48]))

beep(sound = "fanfare")
```

The indexes and adapters have been removed from the sequence files, but the original primers are still attached. Need to account for their presence in the trim amounts and remove them.

12S Forward primer: 21 - GTCGGTAAAACTCGTGCCAGC

12S Reverse primer: 27 - CATAGTGGGGTATCTAATCCCAGTTTG

BF2 primer length: 20 -  GCHCCHGAYATRGCHTTYCC 

BR2 primer length: 20 -  TCDGGRTGNCCRAARAAYCA 


Quality notes:

12 Forward: cut after 220, but caught 2 controls

12 Reverse: cut after 220

BF2 Forward: cut after 240

BR2 Reverse: these are a hot mess. Run aggregates

```{r aggregatetesting}
plotQualityProfile(fns12Fisolate[1:10], aggregate = T)

plotQualityProfile(fns12Risolate, aggregate = T)

plotQualityProfile(fnsBFisolate, aggregate = T)

plotQualityProfile(fnsBRisolate, aggregate = T)

beep(sound = "fanfare")

```

# Have not localized code from here on

# Begin filtering process

First set directories for a new set of folders - the locations that the filtered files will go into

```{r filterlocale}
path_filt12F <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/12S Forward"
path_filt12R <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/12S Reverse"

path_filtBF <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/BF Forward"
path_filtBR <- "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/BF Reverse"
```


Going to do this one at a time cause I'm scared

Running the reads separate even when I do have pairs because the BF2/BR2 has a mismatch in number of files currently and I'd rather do them all the same. From my understanding, this may let some lower quality reads through since it'll allow individual reads pass when it wouldn't pass as a pair, but I figure the trim settings will already take care of most of the issues, and there are further cleaning steps too.

```{r filt_12F}
filtout12F <- filterAndTrim(file.path(path_raw12F, fastqs12F), file.path(path_filt12F, paste0(names12F, "filt.fastq.gz")), trimLeft = 21, truncLen = 220, maxEE = 2, maxN = 0, verbose = TRUE)

saveRDS(filtout12F, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12F_FilteredOutputs.rds")
```
PNW23_BOWU_MR1 did not have any reads left after the filter, file was not written. Image saved in output folder

## NTS - a bit concerned why the control files are still progressing. Might run separate quality profiles of them later. Halt the rest. Proceeding the next day. The controls are very full (not good), but need to proceed regardless. Will spend until March 5th working on this set to figure out if the data is usable, then either make decision on continuing it or switch to AMC. PNW has fewer samples and processing will take less time, so use it as an exercise.


```{r filttroubleshooting}
plotQualityProfile("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/12S/Read1/PNW23_BOWU_MR1.fastq.gz")
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/FiltElim_PNW23_BOWU_MR1.png", plot = plotQualityProfile("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/12S/Read1/PNW23_BOWU_MR1.fastq.gz"))

```


```{r filt_12R}
filtout12R <- filterAndTrim(file.path(path_raw12R, fastqs12R), file.path(path_filt12R, paste0(names12R, "filt.fastq.gz")), trimLeft = 27, truncLen = 200, maxEE = 2, maxN = 0, verbose = TRUE)

saveRDS(filtout12R, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12R_FilteredOutputs.rds")
```


```{r filt_BF}
filtoutBF <- filterAndTrim(file.path(path_rawBF, fastqsBF), file.path(path_filtBF, paste0(namesBF, "filt.fastq.gz")), trimLeft = 20, truncLen = 270, maxEE = 2, maxN = 0, verbose = TRUE)

saveRDS(filtoutBF, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF_FilteredOutputs.rds")
```


```{r filt_BR}
filtoutBR <- filterAndTrim(file.path(path_rawBR, fastqsBR), file.path(path_filtBR, paste0(namesBR, "filt.fastq.gz")), trimLeft = 20, truncLen = 240, maxEE = 2, maxN = 0, verbose = TRUE)

saveRDS(filtoutBR, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BR_FilteredOutputs.rds")
```

# Explore filtered output 

Can look at the dimensions of each RDS output, and order by the filtered read amount, and/or look at the number of total raw reads compared to the filtered out reads

```{r assess filt_12F}
filtout12F
dim(filtout12F)

# Order by filtered read amount
filtout12F[order(filtout12F[,2], decreasing=FALSE),]

# Compare total raw in and filtered out read amounts
colSums(filtout12F)
```

The controls of concern went from 87249 and 78352 raw reads to 1461 and 1700 filtered reads. Still higher than 5 other 'real' samples, but my instinct is that the high raw read amount was due to high remaining numbers of primer dimers left from the sequencing process.

```{r assess filt_12R}
filtout12R
dim(filtout12R)

# Order by filtered read amount
filtout12R[order(filtout12R[,2], decreasing=FALSE),]

# Compare total raw in and filtered out read amounts
colSums(filtout12R)
```

### NTS: Double check exactly how to trim off the primers on the reverse reads. These filtered read numbers are suspiciously high

```{r assess filt_BF}
filtoutBF
dim(filtoutBF)

# Order by filtered read amount
filtoutBF[order(filtoutBF[,2], decreasing=FALSE),]

# Compare total raw in and filtered out read amounts
colSums(filtoutBF)
```

### NTS: Check the trim parameters for the BF2/BR2 samples. It isn't filtering out as much as the 12S samples. Could be appropriate but also maybe not.

```{r assess filt_BR}
filtoutBR
dim(filtoutBR)

# Order by filtered read amount
filtoutBR[order(filtoutBR[,2], decreasing=FALSE),]

# Compare total raw in and filtered out read amounts
colSums(filtoutBR)
```

# Look at filtered trends
```{r filtertrends}
filttrend12F <- ggplot(as.data.frame(filtout12F)) + geom_point(aes(row.names(filtout12F), reads.in), color = "blue") + geom_point(aes(row.names(filtout12F), reads.out), color = "orange") + ggtitle("Filter Trends for PNW 12S Forward Reads")

filttrend12R <- ggplot(as.data.frame(filtout12R)) + geom_point(aes(row.names(filtout12R), reads.in), color = "blue") + geom_point(aes(row.names(filtout12R), reads.out), color = "orange") + ggtitle("Filter Trends for PNW 12S Reverse Reads")

filttrendBF <- ggplot(as.data.frame(filtoutBF)) + geom_point(aes(row.names(filtoutBF), reads.in), color = "blue") + geom_point(aes(row.names(filtoutBF), reads.out), color = "orange") + ggtitle("Filter Trends for PNW BF2/BR2 Forward Reads")

filttrendBR <- ggplot(as.data.frame(filtoutBR)) + geom_point(aes(row.names(filtoutBR), reads.in), color = "blue") + geom_point(aes(row.names(filtoutBR), reads.out), color = "orange") + ggtitle("Filter Trends for PNW BF2/BR2 Reverse Reads")


ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/FilterTrends_12F.png", filttrend12F)
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/FilterTrends_12R.png", filttrend12R)
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/FilterTrends_BF.png", filttrendBF)
ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/FilterTrends_BR.png", filttrendBR)
```

just out of curiosity's sake, let's also explore some quality profiles again from the filtered reads

```{r plotfiltered}
plotQualityProfile("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/12S Forward/PNW22_C914_MR1filt.fastq.gz")

plotQualityProfile("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/12S Forward/PNW23_C523_MR1filt.fastq.gz")

#Compare post and pre-filter and trim

plotQualityProfile("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Filtered/12S Reverse/PNW22_MATD_MR2filt.fastq.gz")

plotQualityProfile("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/Raw/12S/Read2/PNW22_MATD_MR2.fastq.gz")
```

# Error Rates and Dereplication

First we need to set a seed and make sure the paths are all still set.

```{r error setup}
set.seed(4350)

# Set the names for the filtered files we'll be using

# create a list of files in the path
filtnames12Finter <- list.files(path_filt12F, full.names = TRUE)

# Specify I only want the files with the .gz extension
filtnames12F <- filtnames12Finter[grepl('.gz$', filtnames12Finter)]

# Extract just the file name, not the path, and remove the .gz extension. This leaves on the .fastq extension
fastqfilt12F <- tools::file_path_sans_ext(basename(filtnames12F))

# Remove the .fastq extension
names12Ffilt <- tools::file_path_sans_ext(basename(fastqfilt12F))



# Repeat for the other folders

# 12Reverse
filtnames12Rinter <- list.files(path_filt12R, full.names = TRUE)
filtnames12R <- filtnames12Rinter[grepl('.gz$', filtnames12Rinter)]
fastqfilt12R <- tools::file_path_sans_ext(basename(filtnames12R))
names12Rfilt <- tools::file_path_sans_ext(basename(fastqfilt12R))

# BForward
filtnamesBFinter <- list.files(path_filtBF, full.names = TRUE)
filtnamesBF <- filtnamesBFinter[grepl('.gz$', filtnamesBFinter)]
fastqfiltBF <- tools::file_path_sans_ext(basename(filtnamesBF))
namesBFfilt <- tools::file_path_sans_ext(basename(fastqfiltBF))

# BReverse
filtnamesBRinter <- list.files(path_filtBR, full.names = TRUE)
filtnamesBR <- filtnamesBRinter[grepl('.gz$', filtnamesBRinter)]
fastqfiltBR <- tools::file_path_sans_ext(basename(filtnamesBR))
namesBRfilt <- tools::file_path_sans_ext(basename(fastqfiltBR))
```

## Dereplication

This step comes from the dada2 workflow (https://bioconductor.org/packages/devel/bioc/vignettes/dada2/inst/doc/dada2-intro.html#learn-the-error-rates), not Sue's code

```{r derep}
derep12F <- derepFastq(filtnames12F, verbose=TRUE)
derep12R <- derepFastq(filtnames12R, verbose=TRUE)
derepBF <- derepFastq(filtnamesBF, verbose = TRUE)
derepBR <- derepFastq(filtnamesBR, verbose = TRUE)

```

Now that the names are set up, we can run the simulations to learn and estimate the number of errors in each filtered folder. This step is in Sue's code (and is the source of the saveRDS and plotErrors lines), but using the default parameters from the dada2 workflow

## Error Rates

```{r errorrates}
err12F <- learnErrors(derep12F, multithread = FALSE, randomize = TRUE)
# The default nbases is 1e8. Using this default since Sue's 1e6 parameter filled up after only one sample

saveRDS(err12F, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12Forward_Error.rds")

ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12Forward_ErrorPlot.png", plotErrors(err12F, nominalQ = TRUE)) 



err12R <- learnErrors(derep12R, multithread = FALSE, randomize = TRUE)

saveRDS(err12R, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12Reverse_Error.rds")

ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12Reverse_ErrorPlot.png", plotErrors(err12R, nominalQ = TRUE)) 



errBF <- learnErrors(derepBF, multithread = FALSE, randomize = TRUE)

saveRDS(errBF, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BForward_Error.rds")

ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BForward_ErrorPlot.png", plotErrors(errBF, nominalQ = TRUE)) 



errBR <- learnErrors(derepBR, multithread = FALSE, randomize = TRUE)

saveRDS(errBR, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BReverse_Error.rds")

ggsave("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BReverse_ErrorPlot.png", plotErrors(errBR, nominalQ = TRUE)) 

# This takes a long time. If you have to do it again, include this:
beep(sound = "fanfare")
```


# Infer sample composition
```{r samplecomposition}
dada12F <- dada(derep12F, err = err12F, mulithread = FALSE)
print("dada 12F finished")
beep(sound = "coin")

dada12R <- dada(derep12R, err = err12R, mulithread = FALSE)
print("dada 12R finished")
beep(sound = "coin")

dadaBF <- dada(derepBF, err = errBF, mulithread = FALSE)
print("dada BF finished")
beep(sound = "coin")

dadaBR <- dada(derepBR, err = errBR, mulithread = FALSE)
print("dada BR finished")
beep(sound = "fanfare")
```

Now make a sequence table

```{r seqtab}
seqtab12F <- makeSequenceTable(dada12F)
dim(seqtab12F)
saveRDS(seqtab12F, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12F Sequence Table.rds")

seqtab12R <- makeSequenceTable(dada12R)
dim(seqtab12R)
saveRDS(seqtab12R, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12R Sequence Table.rds")

seqtabBF <- makeSequenceTable(dadaBF)
dim(seqtabBF)
saveRDS(seqtabBF, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF Sequence Table.rds")

seqtabBR <- makeSequenceTable(dadaBR)
dim(seqtabBR)
saveRDS(seqtabBR, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BR Sequence Table.rds")

```

# Remove Chimeras
```{r nochim}
seqtab12F.nochim <- removeBimeraDenovo(seqtab12F, method = "consensus", multithread = FALSE, verbose = TRUE)
dim(seqtab12F.nochim)
saveRDS(seqtab12F.nochim, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12F Sequence No Chimera Table.rds")

seqtab12R.nochim <- removeBimeraDenovo(seqtab12R, method = "consensus", multithread = FALSE, verbose = TRUE)
dim(seqtab12R.nochim)
saveRDS(seqtab12R.nochim, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12R Sequence No Chimera Table.rds")

seqtabBF.nochim <- removeBimeraDenovo(seqtabBF, method = "consensus", multithread = FALSE, verbose = TRUE)
dim(seqtabBF.nochim)
saveRDS(seqtabBF.nochim, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BF Sequence No Chimera Table.rds")

seqtabBR.nochim <- removeBimeraDenovo(seqtabBR, method = "consensus", multithread = FALSE, verbose = TRUE)
dim(seqtabBR.nochim)
saveRDS(seqtabBR.nochim, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/BR Sequence No Chimera Table.rds")

beep(sound = "coin")
```

### NTS: decide whether or not to do the QA imaging step (DNASeqAnalysis_2.rmd file, before the negative control)

# Set Negative Control

The process from here on is to download a reference sequence library in .fa.gz format, then use phyloseq to assignTaxonomy to the entire seqtab.nochim objects. Then identify which species are present in the negative controls, and subtract them from the samples (either all samples or batches). This is the decontam process in DNASeqAnalysis_3.rmd. We will subtract the control species from batches of samples, based on which controls were collected, filtered, and extracted with which 'real samples.' Refer to the field and lab notebooks to generate this list. This will probably get messy and might even cross projects. 

[] Use this working PNW code to progress the AMC repo 
[] Generate list of which control samples will act as negative sequence controls for which samples (not project specific)
[] Clean species list and download reference libraries for the 12S primers and BF2/BR2 COI primers
[] AssignTaxonomy
[] Proceed with the decontam process
[] Remove any unwanted taxa
[] Continue with rarefaction analysis and then the rest of the diversity and abundance tests





