---
title: "AMCJoint_Analysis"
author: "BYDavis"
date: "2024-06-26"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


I still want to remove bacterial and plant/algae identifications, so don't save the object yet

```{r remove_taxa, echo = FALSE}
# Remove plant and others
ps12S <- ps_clean %>% 
  subset_taxa(Phylum != "Bacillota") 
ps12S <- ps12S %>% 
  subset_taxa(Phylum != "Chlorophyta")
ps12S <- ps12S %>% 
  subset_taxa(Phylum != "Cyanobacteriota")
ps12S <- ps12S %>% 
  subset_taxa(Phylum != "Pseudomonadota")
ps12S <- ps12S %>% 
  subset_taxa(Phylum != "Foraminifera")
ps12S <- ps12S %>% 
  subset_taxa(Phylum != "Streptophyta")
```

```{r clean_columns, echo = FALSE}
ps12S <- prune_taxa(taxa_sums(ps12S) > 0, ps12S) 
ps12S <- prune_samples(sample_sums(ps12S) > 0, ps12S)

ps12S.stack <- transform_sample_counts(ps12S, function(x) x / sum(x) )

# Plot based on Class
ps12.class <- plot_bar(ps12S.stack, fill="Class") + 
  ggtitle("12S Identified Clean Reads") +
  labs(x = "Sample", y = "ASV Proportion") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) +
  theme(axis.title.x = element_blank())

ps12.class


ggsave("CleanedClassIDs.png", path = path_outputs, ps12.class, width = 8, height = 5, units = "in") 

saveRDS(ps12S, "G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12S/6-Jun03-2024/CleanPhyloseq.RDS")
```

```{r}
# Cleaned Phyloseq Object
psCOI <- readRDS("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/COIB/5-May03-2024/CleanPhyloseq.RDS")
psCOI <- subset_samples(psCOI, Treatment%in%c("LWA"))

# Stacked Bar Function for Cleaned Phyloseq
psCOI.stack <- transform_sample_counts(psCOI, function(x) x / sum(x) )

# Cleaned Phyloseq Object
ps12 <- readRDS("G:/My Drive/2_UMaine FSM - Field Projects/AMC/Data/dataoutputs/12S/6-Jun03-2024/CleanPhyloseq.RDS")
ps12 <- subset_samples(ps12, Treatment%in%c("LWA"))

# Stacked Bar Function for Cleaned Phyloseq
ps12.stack <- transform_sample_counts(ps12, function(x) x / sum(x) )

# Use the following for cleaner plots - pay attention to taxonomic rank
psCOI.class = tax_glom(psCOI, taxrank="Class", NArm=FALSE)
ps12.class = tax_glom(ps12, taxrank="Class", NArm=FALSE)

psCOI.fam = tax_glom(psCOI, taxrank="Family", NArm=FALSE)
ps12.fam = tax_glom(ps12, taxrank="Family", NArm=FALSE)

psCOI.sp = tax_glom(psCOI, taxrank="Species", NArm=FALSE)
ps12.sp = tax_glom(ps12, taxrank="Species", NArm=FALSE)

meltCOI <- melt_phyloseq(psCOI)
melt12 <- melt_phyloseq(ps12)
meltCOIun <- subset(meltCOI, Abundance > "0")
melt12un <- subset(melt12, Abundance > "0")

psCOI <- psCOI %>% 
  subset_taxa(Species != "Homo sapiens")
psCOI <- prune_taxa(taxa_sums(psCOI) > 0, psCOI) 
psCOI <- prune_samples(sample_sums(psCOI) > 0, psCOI)

ps12 <- ps12 %>% 
  subset_taxa(Species != "Homo sapiens")

ps12 <- prune_taxa(taxa_sums(ps12) > 0, ps12) 
ps12 <- prune_samples(sample_sums(ps12) > 0, ps12)

```

# Goals

Exploratory rarefaction
```{r rarecurve, echo = FALSE}
rc12 <-as.data.frame(otu_table(ps12))
test12 <- rarecurve(rc12, step = 10, cex=0.5, label = FALSE)

raremax <- max(rowSums(otu_table(ps12))) # Max SV number is 74165
raremin <- min(rowSums(otu_table(ps12))) # Min SV number is 7
rowSums(otu_table(ps12))


rcC <-as.data.frame(otu_table(psCOI))
testC <- rarecurve(rcC, step = 10, cex=0.5, label = FALSE)

raremaxC <- max(rowSums(otu_table(psCOI))) # Max SV number is 5307
rareminC <- min(rowSums(otu_table(psCOI))) # Min SV number is 0
rowSums(otu_table(psCOI))


# Since we trimmed non-LWA sites out of the object (in config), get an updated snapshot of the phyloseq dimensions
ps12
psCOI
raremax
raremin
raremaxC
rareminC
```

ps12
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 113 taxa and 43 samples ]
sample_data() Sample Data:       [ 43 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 113 taxa by 6 taxonomic ranks ]

psCOI
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 123 taxa and 25 samples ]
sample_data() Sample Data:       [ 28 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 123 taxa by 6 taxonomic ranks ]


Exploratory alpha diversity - not actually very interested in it beyond graphs and dummy check
Plots of taxa at sites - break apart by year 

## Basic exploratories - SV sums, Gamma diversity between east and west
28 COI samples survived past removing unwanted taxa
43 12S samples survived past removing unwanted taxa
```{r}
SVsum_COI <- sample_sums(psCOI)
write.table(SVsum_COI, file.path(path_outputs, "AMCSVsum_COI.txt"))

SVsum_12 <- sample_sums(ps12)
write.table(SVsum_12, file.path(path_outputs, "AMCSVsum_12.txt"))

# 12S total unique counts
length(unique(melt12un$Species))
length(unique(melt12un$Family))
length(unique(melt12un$OTU))
# 26 unique species, 19 unique families, 113 unique OTUs
AMCSpec_12 <- unique(melt12un$Species)
write.table(AMCSpec_12, file.path(path_outputs, "AMCSpec_12Total.txt"))

# COI total unique counts
length(unique(meltCOIun$Species))
length(unique(meltCOIun$Family))
length(unique(meltCOIun$OTU))
# 72 unique species detected, with 123 unique OTUs, across 41 unique families
AMCSpec_COI <- unique(meltCOIun$Species)
write.table(AMCSpec_COI, file.path(path_outputs, "AMCSpec_COITotal.txt"))


# Graph
ggsave("AMC12S_Gamma.png", path = path_outputs, plot_bar(ps12.fam, fill="Family") + facet_grid(~Year, scales= "free_x", space = "free_x") + theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + labs(x = "Site", y = "Number of ASVs", title = "12S Overall Diversity")  + scale_colour_brewer(palette = "Set1") + theme(legend.position = "bottom"), width = 8, height = 7, units = "in") 

ggsave("AMCCOI_Gamma.png", path = path_outputs, plot_bar(psCOI.fam, fill="Family") + facet_grid(~Year, scales= "free_x", space = "free_x") + theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + labs(x = "Site", y = "Number of ASVs", title = "COI Overall Diversity")  + scale_colour_brewer(palette = "Set1") + theme(legend.position = "bottom"), width = 8, height = 10, units = "in") 
```

Gamma by year
```{r}
# Look specifically at 2022 vs 2023
# 12S 2022 unique counts
length(unique(melt12un$Species[melt12un$Year == "2022"]))
length(unique(melt12un$Family[melt12un$Year == "2022"]))
length(unique(melt12un$OTU[melt12un$Year == "2022"]))
# 21 unique species detected, with 56 unique OTUs, across 15 unique families
AMCSpec_12_2022 <- unique(melt12un$Species[melt12un$Year == "2022"])
write.table(AMCSpec_12_2022, file.path(path_outputs, "AMCSpec_12_2022.txt"))


# 12S 2023 unique counts
length(unique(melt12un$Species[melt12un$Year == "2023"]))
length(unique(melt12un$Family[melt12un$Year == "2023"]))
length(unique(melt12un$OTU[melt12un$Year == "2023"]))
# 22 unique species detected, with 87 unique OTUs, across 15 unique families
AMCSpec_12_2023 <- unique(melt12un$Species[melt12un$Year == "2023"])
write.table(AMCSpec_12_2023, file.path(path_outputs, "AMCSpec_12_2023.txt"))


AMC12_Yeardif <- setdiff(AMCSpec_12_2022, AMCSpec_12_2023)
write.table(AMC12_Yeardif, file.path(path_outputs, "AMC12_Yeardif.txt"))
# 5 species detected in 2022 were not detected in 2023


AMC12_Yeardifrev <- setdiff(AMCSpec_12_2023,AMCSpec_12_2022)
write.table(AMC12_Yeardif, file.path(path_outputs, "AMC12_Yeardifrev.txt"))
AMC12_Yeardifrev

# 6 species detected in 2023 were not detected in 2022



# COI 2022 unique counts
length(unique(meltCOIun$Species[meltCOIun$Year == "2022"]))
length(unique(meltCOIun$Family[meltCOIun$Year == "2022"]))
length(unique(meltCOIun$OTU[meltCOIun$Year == "2022"]))
# 33 unique species detected, with 44 unique OTUs, across 22 unique families
AMCSpec_COI_2022 <- unique(meltCOIun$Species[meltCOIun$Year == "2022"])
write.table(AMCSpec_COI_2022, file.path(path_outputs, "AMCSpec_COI_2022.txt"))


# COI 2023 unique counts
length(unique(meltCOIun$Species[meltCOIun$Year == "2023"]))
length(unique(meltCOIun$Family[meltCOIun$Year == "2023"]))
length(unique(meltCOIun$OTU[meltCOIun$Year == "2023"]))
# 64 unique species detected, with 105 unique OTUs, across 36 unique families
AMCSpec_COI_2023 <- unique(meltCOIun$Species[meltCOIun$Year == "2023"])
write.table(AMCSpec_COI_2023, file.path(path_outputs, "AMCSpec_COI_2023.txt"))


AMCCOI_Yeardif <- setdiff(AMCSpec_COI_2022, AMCSpec_COI_2023)
write.table(AMCCOI_Yeardif, file.path(path_outputs, "AMCCOI_Yeardif.txt"))
# 9 species detected in 2022 were not detected in 2023


AMCCOI_Yeardifrev <- setdiff(AMCSpec_COI_2023,AMCSpec_COI_2022)
write.table(AMCCOI_Yeardif, file.path(path_outputs, "AMCCOI_Yeardifrev.txt"))
# 41 species detected in 2023 were not detected in 2022
```

## Plot Alpha Diversity
```{r eDNA}

richCOI <-  estimate_richness(psCOI, measures=c("Chao1", "Shannon", "Simpson"))
write.table(richCOI, file.path(path_outputs, "AMCCOI_AlphaDiversity.txt"))

rich12 <-  estimate_richness(ps12, measures=c("Chao1", "Shannon", "Simpson"))
write.table(rich12, file.path(path_outputs, "AMC12_AlphaDiversity.txt"))


# Plot alpha diversity metrics
richCOIp <- plot_richness(psCOI, x = "TreatType", color = "TreatType", measures = c("Chao1", "Shannon", "Simpson")) + geom_boxplot()  + theme_bw() + labs(title = "COI Alpha Diversity Measures", x = "LWA Treatment Type") + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none") + theme(axis.text = element_text(size = 12))

rich12p <- plot_richness(ps12, x = "TreatType", color = "TreatType", measures = c("Chao1", "Shannon", "Simpson")) + geom_boxplot()  + theme_bw() + labs(title = "12S Alpha Diversity Measures", x = "LWA Treatment Type") + theme(axis.title.y = element_blank()) + theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")  + theme(axis.text = element_text(size = 12)) 
 

# Make a COI richness df with metadata columns and add richness
meltCOImeta <- meltCOIun[ , c('Sample', 'TreatType','LWAPosition','Year')]
meltCOImeta <- unique.data.frame(meltCOImeta)

AlphaCOImeta <-  estimate_richness(psCOI, measures=c("Chao1", "Shannon", "Simpson")) %>%
  rownames_to_column(var = "Sample") %>%
  left_join(meltCOImeta, by = "Sample") 


kruskal.test(Chao1 ~ TreatType, data = AlphaCOImeta) # 0.04521
kruskal.test(Shannon ~ TreatType, data = AlphaCOImeta) # 0.02849
kruskal.test(Simpson ~ TreatType, data = AlphaCOImeta) # 0.05301

# Make a 12S richness df with metadata columns and add richness
melt12meta <- melt12un[ , c('Sample', 'TreatType','LWAPosition','Year')]
melt12meta <- unique.data.frame(melt12meta)

Alpha12meta <-  estimate_richness(ps12, measures=c("Chao1", "Shannon", "Simpson")) %>%
  rownames_to_column(var = "Sample") %>%
  left_join(melt12meta, by = "Sample") 


kruskal.test(Chao1 ~ TreatType, data = Alpha12meta) # 0.0001413
kruskal.test(Shannon ~ TreatType, data = Alpha12meta) # 0.002415
kruskal.test(Simpson ~ TreatType, data = Alpha12meta) # 0.005609

richCOIp

# Save plots
ggsave("AMCCOI_AlphaDiversity.png", path = path_outputs, plot = richCOIp, width = 10, height = 5, units = "in")
ggsave("AMC12S_AlphaDiversity.png", path = path_outputs, plot = rich12p, width = 10, height = 5, units = "in")
```

# Beta Diversity

From AJR:
```{r setup}
# First normalize data so they represent proportions instead of messy counts
psCOI_norm <- transform_sample_counts(psCOI, function(x) x / sum(x) )
ps12_norm <- transform_sample_counts(ps12, function(x) x / sum(x) )

```

```{r PCOA}
# COI

# set up PCOA
JacCOIP <- ordinate(psCOI_norm, method ="PCoA", "jaccard", binary = TRUE)
JacPCType  <- plot_ordination(psCOI_norm, JacCOIP, type="samples", color="TreatType") 
JacPCPos  <- plot_ordination(psCOI_norm, JacCOIP, type="samples", color="LWAPosition") 

## Plot based on TreatType
ggsave("AMCCOI_JaccPCOA_Type.png", path = path_outputs, JacPCType + stat_ellipse(geom = "polygon", type = "norm", alpha = 0.4, level = 0.5, aes(fill = TreatType)) + labs(title = "Jaccard PCOA for AMC COI Data - LWA Type"), width = 6, height = 4, units = "in") 

## Plot based on Position
ggsave("AMCCOI_JaccPCOA_position.png", path = path_outputs, JacPCPos + stat_ellipse(geom = "polygon", type = "norm", alpha = 0.4, level = 0.5, aes(fill = LWAPosition)) + labs(title = "Jaccard PCOA for AMC COI Data - LWA Position"), width = 6, height = 4, units = "in") 

# 12S PCoA prep
Jac12P <- ordinate(ps12_norm, method ="PCoA", "jaccard", binary = TRUE)
JacP12Type <- plot_ordination(ps12_norm, Jac12P, type = "samples", color="TreatType")
JacP12Pos <- plot_ordination(ps12_norm, Jac12P, type = "samples", color="LWAPosition")

# Plot based on TreatType
ggsave("AMC12_JaccPCOA_Type.png", path = path_outputs, JacP12Type + stat_ellipse(geom = "polygon", type = "norm", alpha = 0.4, level = 0.5, aes(fill = TreatType)) + labs(title = "Jaccard PCOA for AMC 12S Data - LWA Type"), width = 6, height = 4, units = "in") 

## Plot based on Position
ggsave("AMC12_JaccPCOA_position.png", path = path_outputs, JacP12Pos + stat_ellipse(geom = "polygon", type = "norm", alpha = 0.4, level = 0.5, aes(fill = LWAPosition)) + labs(title = "Jaccard PCOA for AMC 12S Data - LWA Position"), width = 6, height = 4, units = "in") 

```
Note: we are visualizing the 50% ellipse around the data for each treatment type. 50% is not the significance cuttoff we're using (we're using 5%) but just for visualization purposes we're drawing the central tendency at the 50% mark. the permanova gives a p-value with is much less than 0.05



## Is there a significant difference in stream communities between sites that have undergone LWA restoration and those that have not?
Is there a significant detectable difference in stream communities before and after LWA restoration?
```{r PERMANOVA}
# Set up information for adonis PERMANOVA
JacDistCOI <- distance(psCOI_norm, "jaccard", binary = TRUE)
xCOI <- data.frame(sample_data(psCOI_norm))


JacDist12 <- distance(ps12_norm, "jaccard", binary = TRUE)
x12 <- data.frame(sample_data(ps12_norm))


# Significance tests of interest: 
## Is TreatType significant for both primers? Is Year significant for both primers? Is LWAPosition or Refiltration significant? Does the combination of TreatType and Year or TreatType and Position give more significance than a solo factor?

# COI
# TreatType alone
adonis2(JacDistCOI ~ xCOI$TreatType) # p-value = 0.019 **
adonis2(JacDistCOI ~ xCOI$Year) # p-value = 0.008 *
adonis2(JacDistCOI ~ xCOI$LWAPosition) # p-value = 0.005 *
adonis2(JacDistCOI ~ xCOI$Refiltration) # p-vlue = 0.009 * - however the Refiltration and Year info is nearly identical
adonis2(JacDistCOI ~ TreatType*LWAPosition, data = xCOI) # p-value is 0.623 for the combination, individual factors both significant
adonis2(JacDistCOI ~ TreatType*Year, data = xCOI) # p-value is 0.013 for the combination. Only treatment type significant of individual factors

# Each test runs 999 permutations, so small variance each time, but TreatType is consistently the most significant factor, followed by year and then loosely by LWAPosition 


# 12S
adonis2(JacDist12 ~ x12$TreatType) # 0.007 **
adonis2(JacDist12 ~ x12$Year) # 0.005 **
adonis2(JacDist12 ~ x12$LWAPosition) # 0.003 **
adonis2(JacDist12 ~ x12$Refiltration) # 0.004 **
adonis2(JacDist12 ~ TreatType*LWAPosition, data = x12) # 0.736 for the combination, individual factors still ** significant
adonis2(JacDist12 ~ TreatType*Year, data = x12) # 0.003 for the combination, individual factors both significant
```

Dissimilarity_comparison transfer
```{r}
# make some fake data in the shape of OTU tables ----
set.seed(123)

# `se_matrix` is an otu style table for the SE data
se_matrix <- matrix(rpois(5 * 16, 5), nrow = 5, ncol = 16)

rownames(se_matrix) <- paste("site", 1:5, sep = "_")
colnames(se_matrix) <- paste("family", 1:16, sep = "_")

# `asv_matrix` is the output of dada2/phyloseq for the metabarcoding data
asv_matrix <- se_matrix + rbinom(5 * 16, 8, 0.5) - 4
asv_matrix[asv_matrix < 0] <- 0
asv_matrix <- asv_matrix[-2, ]

# everything below would be the actual analysis you would run

library(vegan)
library(dplyr)
library(tidyr)
library(ggplot2)

# make sure to use the same index (e.g. jaccard) for both data sets
se_jacc <- vegdist(se_matrix, method = "jaccard")
asv_jacc <- vegdist(asv_matrix, method = "jaccard")

# AND MAKE SURE THE SITE NAMES ARE THE SAME BETWEEN DATA SETS!!!!!

# now wrangle the dissimilarity data to use in ggplot

se_jacc_df <- se_jacc |>
  as.matrix() |>
  as.data.frame() |>
  mutate(from = labels(se_jacc)) |>
  pivot_longer(-from, names_to = "to", values_to = "diss_se") |>
  filter(from < to) |>
  mutate(comparison = paste(from, to, sep = ":")) |>
  select(-to, -from)

asv_jacc_df <- asv_jacc |>
  as.matrix() |>
  as.data.frame() |>
  mutate(from = labels(asv_jacc)) |>
  pivot_longer(-from, names_to = "to", values_to = "diss_asv") |>
  filter(from < to) |>
  mutate(comparison = paste(from, to, sep = ":")) |>
  select(-to, -from)

# use inner_join because there might not be full overlap in sites 
# between data sets
jacc_df <- inner_join(asv_jacc_df, se_jacc_df)

# finally plot it
ggplot(jacc_df, aes(x = diss_asv, y = diss_se)) +
  geom_point()

# we can also do a correlation test
cor.test(jacc_df$diss_asv, jacc_df$diss_se)

```


ajr temp transfer
```{r}
# read-in COI
tf <- tempfile(fileext = ".RDS")
d <- drive_download(as_id("1PeLB6Qw_mlu8QKORHdPSS-Or-TLmKSjf"), 
                    path = tf, 
                    overwrite = TRUE)

psCOI <- readRDS(file = d$local_path)
psCOI <- subset_samples(psCOI, Treatment == "LWA")

psCOI_norm <- transform_sample_counts(psCOI, function(x) x / sum(x) )

# remove BLK07 and MB11 because they are outliers
psCOI_mod <- subset_samples(psCOI_norm, 
                            !(rownames(psCOI_norm@otu_table) %in% 
                                c("AMC22_BLK07", "AMC23_MB11")))


# read-in 12S
tf <- tempfile(fileext = ".RDS")
d <- drive_download(as_id("1PGqMF4ylntMXcNTyHQ9KB1t1R2cdMKl8"), 
                    path = tf, 
                    overwrite = TRUE)

ps12 <- readRDS(file = d$local_path)
ps12 <- subset_samples(ps12, Treatment == "LWA")

ps12_norm <- transform_sample_counts(ps12, function(x) x / sum(x) )



# PCOA
JacCOIP <- ordinate(psCOI_mod, method ="PCoA", "jaccard", binary = TRUE)
JacPC <- plot_ordination(psCOI_mod, JacCOIP, type="samples", color="TreatType") 
JacPC + stat_ellipse(geom = "polygon", type = "norm", alpha = 0.4, level = 0.5,
                    aes(fill = TreatType))

# note: we are visualizing the 50% ellipse around the data for each treatment
# type. 50% is not the significance cuttoff we're using (we're using 5%) but 
# just for visualization purposes we're drawing the central tendency at the 50% 
# mark. the permanova gives a p-value with is much less than 0.05

# discussing PCOA graph
# post LWA appears to cluster in top left quad, control and preLWA appear
# clustered in right half...let's do permanova to test significance of that 
# pattern

# PERMANOVA
JacDistCOI <- distance(psCOI_mod, "jaccard", binary = TRUE)
xCOI <- data.frame(sample_data(psCOI_mod))

adonis2(JacDistCOI ~ xCOI$TreatType)

# interpretation of result: significant difference between treatment types (yay!)

# ... repeat above for 12S
Jac12P <- ordinate(ps12_norm, method ="PCoA", "jaccard", binary = TRUE)
JacP12 <- plot_ordination(ps12_norm, Jac12P, type = "samples", color="TreatType")
JacP12 + stat_ellipse(geom = "polygon", type = "norm", alpha = 0.4, level = 0.5,
                      aes(fill = TreatType))

JacDist12 <- distance(ps12_norm, "jaccard", binary = TRUE)
x12 <- data.frame(sample_data(ps12_norm))

adonis2(JacDist12 ~ x12$TreatType)


```