---
title: "AMCNotes"
author: "Beth Davis"
date: "`r Sys.Date()`"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Notes for AMC Control Batching:

List 0 for samples I can't find or link to a control in the meta sheet

batch 1: [ ] 8/26 Lab Control - AMC22_BLK04, ""BLK05, ""BLK06, ""BLK07, ""BLK08

batch 2: [X] 9/28 DI - AMC22_AB01, ""AB04

batch 3: [p] AMC22_C816 - AMC22_C821 - AMC22_IP01 - AMC22_MB07 - AMC22_MB10 - AMC22_BH01 (.1 AND .2)

batch 4: [ ]AMC22_AB03 - AMC22_C803 - AMC22_MB08 - AMC22_BLK05

batch 5: [x] AMC22_C820 - AMC22_WB02 - AMC22_MB01 - AMC22_BLK04 

batch 6: [x] AMC22_IS02 - AMC22_MB06 - AMC22_C814 - AMC22_IS05 - AMC22_MB09 - AMC22_IS04 - AMC22_HB02

batch 7: [x] AMC22_LP01 - AMC22_WBP01_0.5m - AMC22_BLK08 - AMC22_C806 - AMC22_HB03 - AMC22_C811 - AMC22_BLK06 - AMC22_BLK03

batch 8: [x] AMC22_BLK03 - AMC22_WP01_0M - AMC22_HB01 - AMC22_BLK01 - AMC22_C822U - AMC22_BLK02 - AMC22_MOB01 - AMC22_C815 - AMC22_WBP01_1M - AMC22_C804 - AMC22_BLK07 - AMC22_C825U


batch 9: [x] AMC23_C713 - AMC23_MB07 - AMC23_MB09 - AMC23_MB13 - AMC23_MB06 - AMC23_MB11

batch 10: [x] AMC23_C715 - AMC23_MB11 - AMC23_MB12 - AMC23_MB10 - AMC23_MB08 - AMC23_AB03 - AMC23_NE07

batch 11: [x] AMC23_C719 - AMC23_AB02 - AMC23_NE04

batch 12: [x] AMC23_C721 - AMC23_AB04 - AMC23_AB01 | AMC23_C721M - AMC23_SU04 - AMC23_SU02 - AMC23_SU_03

batch 13: [x]AMC23_C724 - AMC23_HB03 - AMC23_BB02 - AMC23_BB03

batch 14: [x] AMC23_C725 - AMC23_BB05 - AMC23_HB03 - AMC23_BB02 - AMC23_HB02 

batch 15: AMC23_C728 - PIN23_C728_W1 - AMC23_MB01 - AMC23_HB05 - PIN23_MOB1_W1 - PIN23_SAS5_W1 - PIN23_MAS2_W1 - PIN23_MLO1_W1 - PIN23_JAB1_W1


# EXTRACTION
AMC22_C822U - AMC22_BLK07 - AMC22_WBP01_1M - AMC22_BHO1.1

AMC23_C728 - AMC23_HB02 - AMC23_BB04 - AMC23_HB03 - AMC23_AB01 - AMC23_SU04

AMC23_C721 - AMC23_C721M - AMC23_C719 - AMC23_MB07 - AMC23_NE07 - AMC23_MB08 - ""SU03 - ""SU02 - ""AB04

AMC23_C724 - ""C713 - ""C715 - ""MB12 - ""MB10 - ""NE04 - ""AB02 - ""MB06 - ""MB13 - ""AB03 - ""MB11 - ""MB09

# 12S Analysis Notes
Error rate seeds:
seed on 3/22: 0743
seed on 5/2: 9450

On merging:
3/22 note:
* Apparently if you filter and trim separately, you cannot merge after that point, and must filterandTrim the forward and reverse reads together. (source: the DADA2 pipeline tutorial)
Restarted filter parameters with that in mind

# COI Analysis Notes

## Filtering

### NTS: 3/05/2024
Parameters used for original COIB filterAndTrim:
filtoutBF <- filterAndTrim(file.path(path_rawBF, fastqsBF), file.path(path_filtBF, paste0(namesBF, "filt.fastq.gz")), trimLeft = 20, truncLen = 270, maxEE = 2, maxN = 0, verbose = TRUE)
filtoutBR <- filterAndTrim(file.path(path_rawBR, fastqsBR), file.path(path_filtBR, paste0(namesBR, "filt.fastq.gz")), trimLeft = 20, truncLen = 240, maxEE = 2, maxN = 0, verbose = TRUE)

### NTS: 3/20/2024
Parameters used for COIB filterAndTrim:
filterAndTrim(file.path(path_rawBF, fastqsBF), file.path(path_filtBF, paste0(namesBF, "filt.fastq.gz")), trimLeft = 50, truncLen = 240, maxEE = 2, maxN = 0, verbose = TRUE)
filtoutBR <- filterAndTrim(file.path(path_rawBR, fastqsBR), file.path(path_filtBR, paste0(namesBR, "filt.fastq.gz")), trimLeft = 60, truncLen = 200, maxEE = 2, maxN = 0, verbose = TRUE)

### NTS: 3/22/2024. After merging the reads with the quality-guided filtering parameters (above), I'm worried about the large drops in read number, and wondering if the trimming was too aggressive, and sacrificed overlap for quality. Going to rerun the BF2/BR2 filtering, into a different output folder, with the following parameters:

BF2 Forward: cut left 20, minQ = 25
filtoutBF <- filterAndTrim(file.path(path_rawBF, fastqsBF), file.path(path_filtBF, paste0(namesBF, "filt.fastq.gz")), trimLeft = 20, minQ = 25, maxEE = 2, maxN = 0, verbose = TRUE)
BR2 Reverse: cut left 20, minQ = 25
filtoutBR <- filterAndTrim(file.path(path_rawBR, fastqsBR), file.path(path_filtBR, paste0(namesBR, "filt.fastq.gz")), trimLeft = 20, minQ = 25, maxEE = 2, maxN = 0, verbose = TRUE)

This should target only removing the primers and low quality reads, and leave as much of the rest as possible intact for overlap

### NTS: 5/03/2024. After talking to Andy about ^, involving minQ was the exact wrong thing to do. I need to remove the primers, so the trimLeft = 20 will stay, and would like to cut off at a quality score of 25. 
